src/georaster-layer-for-leaflet.ts

```diff
@@ -0,0 +1,592 @@
+
+              // y-axis coordinate of the starting point of the rectangle representing the raster pixel
+              const y = yInTilePixels;
+
+              // how many real screen pixels does a pixel of the sampled raster take up
+              const width = widthOfSampleInScreenPixelsInt;
+              const height = heightOfSampleInScreenPixelsInt;
+
+              if (this.options.customDrawFunction) {
+                this.options.customDrawFunction({
+                  values,
+                  context,
+                  x,
+                  y,
+                  width,
+                  height,
+                  rasterX: xInRasterPixels,
+                  rasterY: yInRasterPixels,
+                  sampleX: w,
+                  sampleY: h,
+                  sampledRaster: tileRasters
+                });
+              } else {
+                const color = this.getColor(values);
+                if (color) {
+                  context.fillStyle = color;
+                  context.fillRect(x, y, width, height);
+                }
+              }
+            }
+          }
+        }
+      }
+
+      done && done(error, tile);
+    }, 0);
+
+    // return the tile so it can be rendered on screen
+    return tile;
+  },
+
+  // method from https://github.com/Leaflet/Leaflet/blob/bb1d94ac7f2716852213dd11563d89855f8d6bb1/src/layer/ImageOverlay.js
+  getBounds: function () {
+    this.initBounds();
+    return this._bounds;
+  },
+
+  getMap: function () {
+    return this._map || this._mapToAdd;
+  },
+
+  _isValidTile: function (coords: Coords) {
+    const crs = this.getMap().options.crs;
+
+    if (!crs.infinite) {
+      // don't load tile if it's out of bounds and not wrapped
+      const globalBounds = this._globalTileRange;
+      if (
+        (!crs.wrapLng && (coords.x < globalBounds.min.x || coords.x > globalBounds.max.x)) ||
+        (!crs.wrapLat && (coords.y < globalBounds.min.y || coords.y > globalBounds.max.y))
+      ) {
+        return false;
+      }
+    }
+
+    const bounds = this.getBounds();
+
+    if (!bounds) {
+      return true;
+    }
+
+    const { x, y, z } = coords;
+
+    const layerBounds = L.latLngBounds(bounds);
+
+    const boundsOfTile = this._tileCoordsToBounds(coords);
+
+    // check given tile coordinates
+    if (layerBounds.overlaps(boundsOfTile)) return true;
+
+    // if not within the original confines of the earth return false
+    // we don't want wrapping if using Simple CRS
+    if (crs === L.CRS.Simple) return false;
+
+    // width of the globe in tiles at the given zoom level
+    const width = Math.pow(2, z);
+
+    // check one world to the left
+    const leftCoords = L.point(x - width, y);
+    leftCoords.z = z;
+    if (layerBounds.overlaps(this._tileCoordsToBounds(leftCoords))) return true;
+
+    // check one world to the right
+    const rightCoords = L.point(x + width, y);
+    rightCoords.z = z;
```

Comment:
`L.point` is of type `Coords` which is simply an d `x` and `y`. What is the `z` value for? Is it needed downstream? I can augment the point type for this scenario in this example if need be.

Reply:
I believe the z value is needed here for _tileCoordsToBounds to convert the point to a rectangle representing the tile.  The z represents the zoom level.  X and Y aren't latitude/longitude in this case, but how many mercator tiles from the left and top the tile represented by the coords is.

Reply:
Cool. Casting L.point to a coord type where Coord simply extends Point by including the z. 

```typescript
const leftCoords = L.point(x - width, y) as Coords;
```